dataflow.py

```diff
@@ -247,9 +247,13 @@ def stop_when_loop(item):
     return stop_when_loop
 
 
-class StopPipeline(Exception): pass
+class StopPipeline       (Exception): pass
+class IncompletePipeError(Exception): pass
```

Comment:
I'm not a fan of naming exceptions `*Error`. For some it is more reasonable than others. For example, I think that `KeyError` is completely misnamed as an error: it is not an error to ask a question to which the answer happens not to be known right now. If the answer *cannot* be known, `ZeroDivisionError` for example, then it would be OK to call it an error.

Broadly speaking, exceptions are messages about some noteworthy situations. I'd avoid calling them errors unless it being an error is a fundamental part of the message.

I'd have to think carefully before deciding about this particular message.